### ISSUE TITLE
Accuracy doesn't work for simple binary classifier

### ISSUE THREAD

**Issue by Richardjmorton** [google/flax#4456]
Hello,

This is quite a simple issue but the function `nnx.metric.Accuracy`  only works for a network that returns more than one output, i.e. 
`super().update(values=(logits.argmax(axis=-1) == labels))`. Not sure if this is by design. 

I was playing around with a simple binary classifier than only returned a single logit per observation. The Accuracy function gives nonsense for this case.

Anyway, might be worth noting in the doc string what the function expects.

**Comment by cgarciae** [comment]
Yeah I think current implementation is for multi class. We could add support for binary classification.

**Comment by mattbahr** [comment]
I'm new to the project and am interested in getting involved. I can take a stab at this if no one else is working on it.
